### ISSUE TITLE
Compiling exawind on Summit

### ISSUE THREAD

**Issue by gantech** [sandialabs/spack-manager#405]
I’m running into some issues with installing exawind-driver on Summit. I'm using this commit of Spack manager

```
commit 57a1a8cf0acd51b8b00dfaf662b2e3fa44963cd4 (HEAD -> main, origin/main, origin/HEAD)
Author: psakievich <psakiev@sandia.gov>
Date:   Thu Nov 3 13:34:05 2022 -0600

    ASAN updates for leak in libc on ascic platform (#398)
```

     OpenFAST can’t seem to get HDF5

```
==> Installing openfast-master-32fo2cjb264pzeycb4yn6n6ijxjevkii
==> No binary for openfast-master-32fo2cjb264pzeycb4yn6n6ijxjevkii found: installing from source
==> No patches needed for openfast
==> openfast: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/cmake-3.21.3-7st3jbsm6qhqw4hoysgdbpw6butsljtp/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/autofs/nccs-svm1_home1/ganesh/exawind/spack-manager/spack/opt/spack/linux-rhel8-ppc64le/gcc-10.2.0/openfast-master-32fo2cjb264pzeycb4yn6n6ijxjevkii' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/autofs/nccs-svm1_home1/ganesh/exawind/spack-manager/spack/opt/spack/linux-rhel8-ppc64le/gcc-10.2.0/openfast-master-32fo2cjb264pzeycb4yn6n6ijxjevkii/lib;/autofs/nccs-svm1_home1/ganesh/exawind/spack-manager/spack/opt/spack/linux-rhel8-ppc64le/gcc-10.2.0/openfast-master-32fo2cjb264pzeycb4yn6n6ijxjevkii/lib64;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/hdf5-1.10.7-ssm3jkqyd3xy46b4nhosfkvrbcjhgzji/lib;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/libxml2-2.9.10-fs2vh7wfmj5eqlljohqzs4dwrtvkjrfi/lib;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/lib;/autofs/nccs-svm1_home1/ganesh/exawind/spack-manager/spack/opt/spack/linux-rhel8-ppc64le/gcc-10.2.0/yaml-cpp-0.6.3-bvzmitar6lbyovyakmwz4qvx4gvni7js/lib;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/zlib-1.2.11-2rg3opablnzvdehrgcygxseka3hrtoxd/lib;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/netlib-lapack-3.9.1-ehzlxfcscrj6q5vzslpicc7woxv6vmx5/lib64' '-DCMAKE_PREFIX_PATH:STRING=/autofs/nccs-svm1_home1/ganesh/exawind/spack-manager/spack/opt/spack/linux-rhel8-ppc64le/gcc-10.2.0/yaml-cpp-0.6.3-bvzmitar6lbyovyakmwz4qvx4gvni7js;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/zlib-1.2.11-2rg3opablnzvdehrgcygxseka3hrtoxd;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/netlib-lapack-3.9.1-ehzlxfcscrj6q5vzslpicc7woxv6vmx5;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/libxml2-2.9.10-fs2vh7wfmj5eqlljohqzs4dwrtvkjrfi;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/hdf5-1.10.7-ssm3jkqyd3xy46b4nhosfkvrbcjhgzji;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/cmake-3.21.3-7st3jbsm6qhqw4hoysgdbpw6butsljtp' '-DBUILD_DOCUMENTATION:BOOL=OFF' '-DBUILD_TESTING:BOOL=OFF' '-DBUILD_SHARED_LIBS:BOOL=ON' '-DDOUBLE_PRECISION:BOOL=ON' '-DUSE_DLL_INTERFACE:BOOL=ON' '-DBUILD_OPENFAST_CPP_API:BOOL=ON' '-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON' '-DBLAS_LIBRARIES:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/netlib-lapack-3.9.1-ehzlxfcscrj6q5vzslpicc7woxv6vmx5/lib64/liblapack.so;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/netlib-lapack-3.9.1-ehzlxfcscrj6q5vzslpicc7woxv6vmx5/lib64/libblas.so' '-DLAPACK_LIBRARIES:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/netlib-lapack-3.9.1-ehzlxfcscrj6q5vzslpicc7woxv6vmx5/lib64/liblapack.so;/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/netlib-lapack-3.9.1-ehzlxfcscrj6q5vzslpicc7woxv6vmx5/lib64/libblas.so' '-DCMAKE_CXX_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicxx' '-DCMAKE_C_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicc' '-DCMAKE_Fortran_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpif90' '-DMPI_CXX_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicxx' '-DMPI_C_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicc' '-DMPI_Fortran_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpif90' '-DHDF5_ROOT:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/hdf5-1.10.7-ssm3jkqyd3xy46b4nhosfkvrbcjhgzji' '-DYAML_ROOT:STRING=/autofs/nccs-svm1_home1/ganesh/exawind/spack-manager/spack/opt/spack/linux-rhel8-ppc64le/gcc-10.2.0/yaml-cpp-0.6.3-bvzmitar6lbyovyakmwz4qvx4gvni7js' '-DHDF5_NO_FIND_PACKAGE_CONFIG_FILE:BOOL=ON' '/autofs/nccs-svm1_home1/ganesh/exawind/spack-manager/stage/spack-stage-openfast-master-32fo2cjb264pzeycb4yn6n6ijxjevkii/spack-src'

3 errors found in build log:
  >> 6     /lib/../lib64/crt1.o:(.rodata+0x8): undefined reference to `main'
     7     /usr/bin/ld: link errors found, deleting executable `a.out'
  >> 8     collect2: error: ld returned 1 exit status
     9     -- Detecting CXX compiler ABI info
     10    -- Detecting CXX compiler ABI info - done
     11    -- Check for working CXX compiler: /sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gc
           c-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicxx - sk
           ipped
     12    -- Detecting CXX compile features
     13    -- Detecting CXX compile features - done
     14    -- Detecting C compiler ABI info

     ...

     30    -- Found MPI_C: /sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-m
           pi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicc (found version "3.1")
     31    -- Found MPI_CXX: /sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum
           -mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicxx (found version "3.1")
     32    -- Found MPI_Fortran: /sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spec
           trum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpif90 (found version "3
           .1")
     33    -- Found MPI: TRUE (found version "3.1")
     34    -- Found LibXml2: /sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/libxml2-
           2.9.10-fs2vh7wfmj5eqlljohqzs4dwrtvkjrfi/lib/libxml2.so (found version "2.9.10")
     35    -- Found ZLIB: /sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/zlib-1.2.11
           -2rg3opablnzvdehrgcygxseka3hrtoxd/lib/libz.so (found version "1.2.11")
  >> 36    CMake Error at /autofs/nccs-svm1_sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-1
           0.2.0/cmake-3.21.3-7st3jbsm6qhqw4hoysgdbpw6butsljtp/share/cmake-3.21/Modules/FindPackage
           HandleStandardArgs.cmake:230 (message):
     37      Could NOT find HDF5 (missing: HDF5_HL_LIBRARIES)
     38    Call Stack (most recent call first):
     39      /autofs/nccs-svm1_sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/cmake-3
           .21.3-7st3jbsm6qhqw4hoysgdbpw6butsljtp/share/cmake-3.21/Modules/FindPackageHandleStandar
           dArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
     40      /autofs/nccs-svm1_sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/cmake-3
           .21.3-7st3jbsm6qhqw4hoysgdbpw6butsljtp/share/cmake-3.21/Modules/FindHDF5.cmake:1001 (fin
           d_package_handle_standard_args)
     41      glue-codes/openfast-cpp/CMakeLists.txt:27 (find_package)
     42

See build log for details:
  /autofs/nccs-svm1_home1/ganesh/exawind/spack-manager/stage/spack-stage-openfast-master-32fo2cjb264pzeycb4yn6n6ijxjevkii/spack-build-out.txt
```

Everything else compiles… just openfast seems to have a problem.

**Comment by lawrenceccheung** [comment]
@gantech, @psakievich, I think I saw the same thing on https://github.com/psakievich/spack-manager/issues/345

**Comment by psakievich** [comment]
#410 should fix this and #345.  @gantech @lawrenceccheung will you try this out and see if the issues are resolved?

**Comment by psakievich** [comment]
Closing since @gantech was able to build the driver on summit
